internal/http/routes/api/v1/auth.go

```diff
@@ -150,6 +163,40 @@ func (r *AuthAPIRoutes) meHandler(c *gin.Context) {
 	response.Send(c, http.StatusOK, ctx.GetAccount())
 }
 
+func (r *AuthAPIRoutes) settingsHandler(c *gin.Context) {
+	var payload settingRequestPayload
+	if err := c.ShouldBindJSON(&payload); err != nil {
+		response.SendInternalServerError(c)
+		return
+	}
+
+	if err := payload.IsValid(); err != nil {
+		response.SendError(c, http.StatusBadRequest, err.Error())
+		return
+	}
+	ctx := context.NewContextFromGin(c)
+	if ctx.UserIsLogged() {
+		response.SendError(c, http.StatusForbidden, nil)
+		return
+	}
```

Comment:
You are not returning anything, really. Just finishing the function early, since this functions doesn't have return values :)

You can move this to the beginning of the function, too. Since you don't need to do anything if the user is not logged in.

Reply:
You need to send the new authentication JWT token with this request for this to work. The auth token is under `shiori-token` in the local storage. Send it as such: `Authorization: Bearer ${token}` and your request should work.